pkg/common/cli/umask.go

```diff
@@ -0,0 +1,43 @@
+package cli
+
+import "github.com/sirupsen/logrus"
+
+// The umask for SPIRE processes should not allow write by group, or
+// read/write/execute by everyone.
+const minimumUmask = 0027
+
+// SetUmask sets the desired umask if desiredUmask is >= 0. The desired umask
+// is upgraded to include the minimumUmask bits if it is weaker. If
+// desiredUmask is < 0, then the current umask is upgraded to include the
+// minimumUmask bits, if necessary.
+func SetUmask(log logrus.FieldLogger, desiredUmask int) {
+	if desiredUmask >= 0 {
+		log.Warn("Setting umask via configuration is deprecated!")
+		if !umaskSupported {
+			log.Warn("Umask not supported on this platform.")
+			return
+		}
+		if (desiredUmask & minimumUmask) != minimumUmask {
+			badUmask := desiredUmask
+			desiredUmask |= minimumUmask
+			log.Warnf("Desired umask %#04o is too permissive; setting umask %#04o.", badUmask, desiredUmask)
+		} else {
+			log.Warnf("Setting umask %#04o.", desiredUmask)
```

Comment:
Maybe we can get away without this one? Or perhaps it could be a debug message?

Reply:
I'd be fine making it debug.

Reply:
Actually, i just consolidated it into the warning message for deprecation. I think it turned out pretty clean.